plugin-nabu/src/main/java/ca/on/oicr/gsi/shesmu/nabu/ArchiveCaseAction.java

```diff
@@ -0,0 +1,247 @@
+            + formatSetAsString(workflowRunIdsForOffsiteArchive)
+            + "], "
+            + "\"workflowRunIdsForVidarrArchival\": ["
+            + formatSetAsString(workflowRunIdsForVidarrArchival)
+            + "]"
+            + "}";
+    return body;
+  }
+
+  private String formatSetAsString(Set<String> set) {
+    return String.join(
+        ",", set.stream().map(name -> ("\"" + name + "\"")).collect(Collectors.toList()));
+  }
+
+  @Override
+  public ActionState perform(
+      ActionServices services, Duration lastGeneratedByOlive, boolean isOliveLive) {
+    final var overloaded = services.isOverloaded("all", "nabu");
+    if (!overloaded.isEmpty()) {
+      this.errors =
+          Collections.singletonList("Overloaded services: " + String.join(", ", overloaded));
+      return ActionState.THROTTLED;
+    }
+
+    HttpRequest.BodyPublisher body;
+    try {
+      body = HttpRequest.BodyPublishers.ofString(createRequestBody());
+    } catch (final Exception e) {
+      e.printStackTrace();
+      this.errors = Collections.singletonList(e.getMessage());
+      return ActionState.FAILED;
+    }
+    final var baseUrl = owner.get().NabuUrl();
+
+    final var request =
+        HttpRequest.newBuilder(URI.create(baseUrl + "/case"))
+            .header("Content-type", "application/json")
+            .header("Accept", "application/json")
+            .POST(body)
+            .build();
+
+    owner.log("NABU REQUEST: " + request, LogLevel.DEBUG, null);
+    try (var timer = NabuRequestTime.start(baseUrl)) {
+      var response =
+          HTTP_CLIENT.send(request, new JsonListBodyHandler<>(MAPPER, NabuCaseArchiveDto.class));
+      if (response.statusCode() == 409) {
+        return ActionState.HALP;
+      } else if (response.statusCode() / 100 != 2) {
+        NabuRequestErrors.labels(baseUrl).inc();
+        showHTTPError(response, baseUrl);
+        return ActionState.FAILED;
+      } else if (response.statusCode() == 201) {
+        return ActionState.INFLIGHT;
+      } else if (response.statusCode() == 200) {
+        final var results = response.body().get().collect(Collectors.toList());
+        return actionStatusFromArchive(results.get(0));
+      } else {
+        return ActionState.FAILED;
+      }
+    } catch (final Exception e) {
+      e.printStackTrace();
+      this.errors = Collections.singletonList(e.getMessage());
+      NabuRequestErrors.labels(baseUrl).inc();
+      return ActionState.FAILED;
+    }
+  }
+
+  @Override
+  public int priority() {
+    return 0;
+  }
+
+  @Override
+  public long retryMinutes() {
+    return 10;
+  }
+
+  private void showHTTPError(HttpResponse<?> response, String url)
+      throws UnsupportedOperationException, IOException {
+    final List<String> errors = new ArrayList<>();
+    final Map<String, String> labels = new TreeMap<>();
+    labels.put("url", url);
+    owner.log("HTTP error: " + response.statusCode(), LogLevel.ERROR, labels);
```

Comment:
This will only log "HTTP error: 418", won't it? Do you want more information than that in the log?

Reply:
I don't think there needs to be more? There's enough info to check nabu for http status code meanings